src/components/street-align.js

```diff
@@ -0,0 +1,99 @@
+/* global AFRAME */
+
+AFRAME.registerComponent('street-align', {
+  dependencies: ['managed-street'],
+  schema: {
+    width: {
+      default: 'center',
+      type: 'string',
+      oneOf: ['center', 'left', 'right']
+    },
+    length: {
+      default: 'start',
+      type: 'string',
+      oneOf: ['middle', 'start', 'end']
+    }
+  },
+
+  init: function () {
+    // Listen for any segment changes from managed-street
+    this.realignStreet = this.realignStreet.bind(this);
+    this.el.addEventListener('segments-changed', this.realignStreet);
+
+    // Initial alignment
+    this.realignStreet();
```

Comment:
remove this call, it's done a second time in the setTimeout

Reply:
1604642c86d0c446fda6c86ca26a539b0b7dd3a8